nexus/inventory/src/collector.rs

```diff
@@ -74,7 +77,7 @@ impl<'a> Collector<'a> {
 
     /// Collect inventory from all MGS instances
     async fn collect_all_mgs(&mut self) {
-        let clients = self.mgs_clients.clone();
+        let clients = self.mgs_clients.take().unwrap();
```

Comment:
Use `.expect("collect_all_mgs called once")` maybe?

Reply:
Done in [cb59c68](https://github.com/oxidecomputer/omicron/pull/6810/commits/cb59c682d2b5843a45a244256b71654fdaa77249)

Reply:
Ha, I realized I said done. But actually I wrote that and removed it as it's no longer necessary with `collect_one_xxx` being static methods.